platforms/software/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPomDeploymentRepository.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven.internal.publication;
+
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Property;
+import org.gradle.api.publish.maven.MavenPomDeploymentRepository;
+
+import javax.inject.Inject;
+
+public class DefaultMavenPomDeploymentRepository implements MavenPomDeploymentRepository {
```

Comment:
This class should not be necessary at all. 

`objectFactory.newInstance(MavenPomDeploymentRepository.class);` should generate a decorated implementation of the interface without needing a concrete default implementation.

Reply:
Okay, should I also create the MavenPomDeploymentRepository instance in the constructor to make the `getRepository` getter not Nullable?

And should I also create another PR to update the `relocation` property too? Or is it okay to do it in this PR?

Reply:
@jvandort What about DefaultMavenPomRelocation?